pkg/server/server.go

```diff
@@ -69,6 +69,10 @@ func New(
 	admissionControlPolicy string,
 	admissionControlPolicyLimitMaxConcurrentTxn int64,
 	admissionControlOnlyWriteQueries bool,
+	datastoreMaxIdleConnections int,
```

Comment:
I think this method was already accepting too many arguments. Can we have an option struct instead? Something like `ServerOptions` or even just `Options`?

Reply:
Agreed, using the struct from generic now.